File: Interpreted/mutual-recursion/mrec.jl
F(n) = n == 0 ?  one(n) : n - M(F(n - 1))
M(n) = n == 0 ? zero(n) : n - F(M(n - 1))

println(F(parse(Int, ARGS[1])));
println(M(parse(Int, ARGS[1])));

File: Interpreted/mutual-recursion/mrec.clj
(declare F) ; forward reference

(defn M [n]
  (if (zero? n)
    0
    (- n (F (M (dec n))))))

(defn F [n]
  (if (zero? n)
    1
    (- n (M (F (dec n))))))

(println (F (Integer/parseInt (first *command-line-args* ))))
(println (M (Integer/parseInt (first *command-line-args* ))))
